src/hocs/ec-with-abortable-fetch/ec-with-abortable-fetch.spec.js

```diff
@@ -165,7 +165,9 @@ describe('EcWithAbortableFetch', () => {
 
   it('should abort previous fetch calls', async () => {
     const dataSource = {
-      fetch: jest.fn().mockResolvedValueOnce({ result: 1 }).mockResolvedValueOnce({ result: 2 }),
+      fetch: jest.fn()
+        .mockImplementationOnce(() => new Promise((resolve) => { setTimeout(() => resolve({ result: 1 }), 1); }))
+        .mockImplementationOnce(() => new Promise((resolve) => { setTimeout(() => resolve({ result: 2 }), 1); })),
```

Comment:
Why did this change? What was the problem?

Reply:
After migrating to Jest 29 `mockResolvedValueOnce` seems to resolved immediately like if they are not the promises and code inside component should await on it. When we change `props` the second time, we are waiting that watcher in the component will fire the `fetch` function before the first `mockResolvedValueOnce` resolved, but that not happens and the second `fetch` is running when the first one is already resolved and `this.fetchAbortController` is null. 

I believe we could remove this hoc and replace it with composable soon?

Reply:
there's roughly 25+ HoCs using this in EBO, so it's going to take some time before we can get rid of it.

The docs in Jest says that 
> mockFn.mockResolvedValueOnce(value)
> Shorthand for:
> jest.fn().mockImplementationOnce(() => Promise.resolve(value));

Can we use it in `Promise.resolve()` form and remove `setTimeout(..., 1)`, because it looks very hacky.

Reply:
I've read that documentation and tried that way, it doesn't work, either.